lib/github.ml

```diff
@@ -235,6 +250,79 @@ let assert_tag_exists ~dry_run tag =
   if Vcs.tag_exists ~dry_run repo tag then Ok ()
   else R.error_msgf "%s is not a valid tag" tag
 
+(* Ask the user then push the tag. Guess the ssh URI from the dev-repo.
+   This function can abort:
+   - The user answered "N" to pushing the tag
+   - The push failed
+
+   This function does nothing if the tag is already present on the remote and
+   point to the same ref. *)
+let push_tag ~dry_run ~yes ~dev_repo vcs tag =
+  let remote_has_tag_uptodate () =
+    Vcs.commit_id ~dirty:false ~commit_ish:tag vcs >>= fun local_rev ->
+    Vcs.ls_remote ~dry_run vcs ~kind:`Tag ~filter:tag dev_repo >>= function
+    | [] -> Ok false
+    | (remote_rev_unpeeled, _) :: _ -> (
+        (* Resolve again in case of annotated commits (most common case).
+           This is a no-op for non-annotated commits. In case of error, we
+           can assume that the remote is different because we checked that we
+           have the tag locally. *)
+        match Vcs.commit_id ~commit_ish:remote_rev_unpeeled vcs with
+        | Ok remote_rev when remote_rev = local_rev ->
+            if remote_rev_unpeeled = remote_rev then
+              App_log.unhappy (fun l ->
+                  l
+                    "The tag present on the remote is not annotated (it was \
+                     not created by dune-release tag.)");
+            Ok true
+        | r ->
+            let pp_r fmt = function
+              | Ok remote_rev -> Text.Pp.commit fmt remote_rev
+              | Error _ -> Format.fprintf fmt "that we don't have locally"
+            in
+            App_log.unhappy (fun l ->
+                l
+                  "The tag %a is present on the remote but points to a \
+                   different commit (%a)."
+                  Text.Pp.version tag pp_r r);
+            Ok false )
```

Comment:
I would have expected an `Error` here. Do you want to try pushing the tag even though the tag already exists on the remote (pointing to a wrong commit)? Does github allow that?

Reply:
The push has `--force`.
No need to fail here, the output will be this warning followed by the question "Push tag ... ?", which would work as usual.

Reply:
Ok, perfect then.